packages/flutter/lib/src/rendering/table.dart

```diff
@@ -0,0 +1,784 @@
+/// horizontal borders and the inner vertical borders.
+class TableBorder extends Border {
+  const TableBorder({
+    BorderSide top: BorderSide.none,
+    BorderSide right: BorderSide.none,
+    BorderSide bottom: BorderSide.none,
+    BorderSide left: BorderSide.none,
+    this.horizontalInside: BorderSide.none,
+    this.verticalInside: BorderSide.none
+  }) : super(
+    top: top,
+    right: right,
+    bottom: bottom,
+    left: left
+  );
+
+  factory TableBorder.all({
+    Color color: const Color(0xFF000000),
+    double width: 1.0
+  }) {
+    final BorderSide side = new BorderSide(color: color, width: width);
+    return new TableBorder(top: side, right: side, bottom: side, left: side, horizontalInside: side, verticalInside: side);
+  }
+
+  factory TableBorder.symmetric({
+    BorderSide inside: BorderSide.none,
+    BorderSide outside: BorderSide.none
+  }) {
+    return new TableBorder(
+      top: outside,
+      right: outside,
+      bottom: outside,
+      left: outside,
+      horizontalInside: inside,
+      verticalInside: inside
+    );
+  }
+
+  final BorderSide horizontalInside;
+
+  final BorderSide verticalInside;
+
+  @override
+  TableBorder scale(double t) {
+    return new TableBorder(
+      top: top.copyWith(width: t * top.width),
+      right: right.copyWith(width: t * right.width),
+      bottom: bottom.copyWith(width: t * bottom.width),
+      left: left.copyWith(width: t * left.width),
+      horizontalInside: horizontalInside.copyWith(width: t * horizontalInside.width),
+      verticalInside: verticalInside.copyWith(width: t * verticalInside.width)
+    );
+  }
+
+  static TableBorder lerp(TableBorder a, TableBorder b, double t) {
+    if (a == null && b == null)
+      return null;
+    if (a == null)
+      return b.scale(t);
+    if (b == null)
+      return a.scale(1.0 - t);
+    return new TableBorder(
+      top: BorderSide.lerp(a.top, b.top, t),
+      right: BorderSide.lerp(a.right, b.right, t),
+      bottom: BorderSide.lerp(a.bottom, b.bottom, t),
+      left: BorderSide.lerp(a.left, b.left, t),
+      horizontalInside: BorderSide.lerp(a.horizontalInside, b.horizontalInside, t),
+      verticalInside: BorderSide.lerp(a.verticalInside, b.verticalInside, t)
+    );
+  }
+
+  @override
+  bool operator ==(dynamic other) {
+    if (super != other)
+      return false;
+    final TableBorder typedOther = other;
+    return horizontalInside == typedOther.horizontalInside &&
+           verticalInside == typedOther.verticalInside;
+  }
+
+  @override
+  int get hashCode => hashValues(super.hashCode, horizontalInside, verticalInside);
+
+  @override
+  String toString() => 'TableBorder($top, $right, $bottom, $left, $horizontalInside, $verticalInside)';
+}
+
+/// Vertical alignment options for cells in [RenderTable] objects.
+///
+/// This is specified using [TableCellParentData] objects on the
+/// [RenderObject.parentData] of the children of the [RenderTable].
+enum CellVerticalAlignment {
```

Comment:
I've made this change, but if we never add horizontal alignment (which I don't think we should or will, really) we should probably remove the word "vertical" from this class and the property, for brevity.